### PR TITLE
topdown/eval: fix 'every' term plugging on save

### PR DESCRIPTION
**Pull request by srenatus**
Previously missing plugging could cause unsafe variables in the
PE output.

Now, all terms in the 'every' body should be plugged properly.
The approach taken here is to plug them all, and then fix the
key and val var names of the copied every expression. Those vars
are fresh after the compiler is done with the expression, so
plugging them should never have any effect outside of the rename.

Fixes _the first part of_ #4766

### CODE REVIEW

File: topdown/eval.go [hunk 1]
```diff
@@ -3044,18 +3042,20 @@ func (e *evalEvery) save(iter unifyIterator) error {
...
-			every.Body[i].Terms = e.e.bindings.Plug(t)
+			every.Body[i].Terms = e.e.bindings.PlugNamespaced(t, e.e.caller.bindings)
 		case []*ast.Term:
-			for j := range t {
-				t[j] = e.e.bindings.Plug(t[j])
+			for j := 1; j < len(t); j++ { // don't plug operator, t[0]
+				t[j] = e.e.bindings.PlugNamespaced(t[j], e.e.caller.bindings)
 			}
 		}
 	}
```
Comment (tsandall) [inline topdown/eval.go:3054]: Unrelated to these changes, but it looks like we're only dealing with single terms and calls here. What happens if the body contains an `ast.Every`? Is there some rewriting that I'm not aware of?

**Review by tsandall — APPROVED** [review]
Since we're saving the body of `every` without PE-ing it we need to namespace the variables to avoid conflicts w/ vars from any the callers. Nice find!

File: topdown/eval.go (hunk 1 above)
Comment (srenatus) [reply topdown/eval.go:3054]: Good find. Probably an oversight. I'll look into it later.

Comment (srenatus) [reply topdown/eval.go:3054]: https://github.com/open-policy-agent/opa/issues/4826

**Pull request merged**